x-pack/plugins/apm/common/agent_configuration/runtime_types/storage_size_rt.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
+import { amountAndUnitToObject } from '../amount_and_unit';
+import { getRangeTypeMessage } from './get_range_type_message';
+
+const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB'];
+
+function toBytes(amount: number, unit: string, decimalUnitBase?: boolean) {
+  const base = decimalUnitBase ? 1000 : 1024;
+  const unitExponent = unit ? units.indexOf(unit.toUpperCase()) : 0;
+  if (unitExponent < 0) {
+    return;
+  }
+  return amount * base ** unitExponent;
+}
+
+function amountAndUnitToBytes(
+  value?: string,
+  decimalUnitBase?: boolean
+): number | undefined {
+  if (value) {
+    const { amount, unit } = amountAndUnitToObject(value);
+    if (isFinite(amount) && unit) {
```

Comment:
You could use [isFiniteNumber](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/common/utils/is_finite_number.ts#L11:L11) instead.